components/style/gecko/conversions.rs

```diff
@@ -483,18 +483,18 @@ pub mod basic_shape {
     // Will panic on NoBox
     // Ideally these would be implemented on Option<T>,
     // but coherence doesn't like that and TryFrom isn't stable
-    impl From<StyleClipPathGeometryBox> for GeometryBox {
-        fn from(reference: StyleClipPathGeometryBox) -> Self {
-            use gecko_bindings::structs::StyleClipPathGeometryBox::*;
+    impl From<StyleGeometryBox> for GeometryBox {
+        fn from(reference: StyleGeometryBox) -> Self {
+            use gecko_bindings::structs::StyleGeometryBox::*;
             match reference {
-                NoBox => panic!("Shouldn't convert NoBox to GeometryBox"),
                 Content => GeometryBox::ShapeBox(ShapeBox::Content),
                 Padding => GeometryBox::ShapeBox(ShapeBox::Padding),
                 Border => GeometryBox::ShapeBox(ShapeBox::Border),
                 Margin => GeometryBox::ShapeBox(ShapeBox::Margin),
                 Fill => GeometryBox::Fill,
                 Stroke => GeometryBox::Stroke,
                 View => GeometryBox::View,
+                _ => panic!("unexpected StyleGeometryBox value"),
```

Comment:
Probably worth printing out the value in the panic message like:

```rust
other => panic!("Unexpected StyleGeometryBox::{:?} while converting to GeometryBox", other),
```

(or as you see fit)